http-client/src/main/java/io/airlift/http/client/HttpUriBuilder.java

```diff
@@ -55,7 +57,7 @@ private HttpUriBuilder(URI previous)
         scheme = previous.getScheme();
         host = previous.getHost();
         port = previous.getPort();
-        path = percentDecode(previous.getRawPath());
+        path = isNull(previous.getRawPath()) ? null : percentDecode(previous.getRawPath());
```

Comment:
I just looked at this code for a quick review before merging and that line confused me. So I'm going to revert to your original version which is much more obvious and straight forward :)